source/dofs/dof_handler.cc

```diff
@@ -1000,7 +1000,7 @@ void DoFHandler<dim,spacedim>::distribute_dofs (const FiniteElement<dim,spacedim
 
   // Only recreate the FECollection if we don't already store
   // the exact same FiniteElement object.
-  if  (fe_collection.size() == 0 || &(fe_collection[0]) != &ff)
```

Comment:
that might not work for FE_Enriched as in the most general case `const std::vector< std::vector< std::function<...>>>` is not encoded in its [name](https://www.dealii.org/developer/doxygen/deal.II/fe__enriched_8cc_source.html#l00429), but i guess it's the problem of FE_Enriched.